homeassistant/components/melnor/models.py

```diff
@@ -71,3 +73,28 @@ def _handle_coordinator_update(self) -> None:
     def available(self) -> bool:
         """Return True if entity is available."""
         return self._device.is_connected
+
+
+class MelnorZoneEntity(MelnorBluetoothBaseEntity):
+    """Base class for valves that define themselves as child devices."""
+
+    _valve: Valve
+
+    def __init__(
+        self,
+        coordinator: MelnorDataUpdateCoordinator,
+        index: int,
+    ) -> None:
+        """Initialize a valve entity."""
+        super().__init__(coordinator)
+
+        self._valve = cast(Valve, self._device[f"zone{index}"])
```

Comment:
It would be nice if the library followed https://peps.python.org/pep-0561/ so you could avoid the cast

Reply:
I can take a look at doing that, sure. I need to do another round of major work to RE the scheduling system off the valve and build support anyway. So I'll be pivoting back to heavy library work after another PR or two.

Reply:
😬 so this is already typed. I guess I had the cast to ignore the `| None` return type. This line has been moving around since April and I hadn't thought twice about it. I'll wrap this up a little better.